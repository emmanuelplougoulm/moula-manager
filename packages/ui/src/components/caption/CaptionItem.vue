<template>
  <span v-if="text" data-testid="caption-item" class="text-small-regular">
    {{ props.text }}
  </span>
</template>

<script setup lang="ts">
const props = defineProps({
  text: {
    type: String,
    default: ''
  }
});
</script>

<style scoped>
span {
  color: var(--neutral-40);
}
</style>
