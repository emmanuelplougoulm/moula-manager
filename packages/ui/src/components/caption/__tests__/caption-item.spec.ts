import { mount } from '@vue/test-utils';
import CaptionItem from '../CaptionItem.vue';

describe('CaptionItem tests', () => {
  it('should display the caption text', () => {
    const wrapper = mount(CaptionItem, {
      props: {
        text: 'This is a caption'
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should not display the caption text', () => {
    const wrapper = mount(CaptionItem, {
      props: {
        text: undefined
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
