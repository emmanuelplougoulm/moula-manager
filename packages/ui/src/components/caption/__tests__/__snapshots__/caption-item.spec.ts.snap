// Vitest Snapshot v1

exports[`CaptionItem tests > should display the caption text 1`] = `"<span data-v-b41f83c4=\\"\\" data-testid=\\"caption-item\\" class=\\"text-small-regular\\">This is a caption</span>"`;

exports[`CaptionItem tests > should not display the caption text 1`] = `"<!--v-if-->"`;
